e2e/test-helpers.go

```diff
@@ -0,0 +1,127 @@
+/*
+Copyright 2024 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/GoogleCloudPlatform/k8s-cloud-provider/pkg/cloud"
+	"github.com/GoogleCloudPlatform/k8s-cloud-provider/pkg/cloud/meta"
+	"github.com/GoogleCloudPlatform/k8s-cloud-provider/pkg/cloud/rgraph/exec"
+)
+
+func expectActions(got []exec.Action, want []exec.ActionType) error {
```

Comment:
// expectActions compares got with want, checking that the list of actions is <???>

What is the predicate that is being checked here?

You need to document all functions that are utility and intended to be used in a common way, even if they are not exported.

This is especially important as the docstring should define a loose sense of what you want the contract of the function to be. Otherwise, people maintaining the code will not understand what you intended and may not make the right choices.

* Does order matter?
* Are there fields that are being ignored? If this is temporary situation, then document a TODO.

Reply:
done